### DISCUSSION TITLE
Another version to Q2 in _2015b94

### DISCUSSION THREAD

**Discussion by tomerab1** [Koren-Ben-Ezra/Exams#6] [Bug Discussions🪲]
Just solved it using a sliding window technique, maybe you'd like to add it as another example of a valid solution.
```  
public static int smallestSub(int[] a, int k) {
        int l = 0, r = 1, sum = a[l];

        while (r < a.length) {
            sum += a[r++];
            while (sum > k) {
                sum -= a[l++];
            }
        }
        return r - l + 1;
    }
```

**Comment by korenbenezra** [comment]


**Reply by tomerab1** [reply]
היעילות זמן ריצה היא: O(n) שכן אנחנו עוברים על המערך פעם אחת בלבד, והלולאה הפנימית משמשת רק על מנת להקטין את גודל "החלון" ולקדם את l.
מבחינת זיכרון כל המשתנים הם קבועים ולכן הסיבוכיות היא O(1).
